daisy/compute/compute.go

```diff
@@ -1026,3 +1038,19 @@ func (c *client) SetCommonInstanceMetadata(project string, md *compute.Metadata)
 
 	return c.i.globalOperationsWait(project, op.Name)
 }
+
+// GetGuestAttributes gets a Guest Attributes.
+func (c *client) GetGuestAttributes(project, zone, name, queryPath, variableKey string) (*computeBeta.GuestAttributes, error) {
+	call := c.rawBeta.Instances.GetGuestAttributes(project, zone, name)
+	if queryPath != "" {
+		call = call.QueryPath(queryPath)
+	}
+	if variableKey != "" {
+		call = call.VariableKey(variableKey)
+	}
+	a, err := call.Do()
+	if shouldRetryWithWait(c.hc.Transport, err, 2) {
```

Comment:
Just curious, is the reason you can't pass a reference to `call.Do` because of the lack of generics?

Also, why would getting guest attributes fail in a way that retries are necessary? Do guest attributes have outages when the VM is running?

Reply:
Pretty much yes, call is different on each of these functions that use shouldRetryWithWait. This is fixable in go with interfaces but that's not how the autogen api is created so we have to deal with it

This retries on error codes that indicate we should retry (the compute api giving us a 5xx error). These errors can be for a number of reasons, but we don't really care other than the api docs say to retry. Actually the docs on what errors we can get are very bad to non existent, thus the all 5xx errors here even though docs say 500 and 503.